src/sentry/eventstream/kafka/protocol.py

```diff
@@ -172,9 +193,18 @@ def get_task_kwargs_for_message_from_headers(
                 "is_regression": is_regression,
                 "is_new_group_environment": is_new_group_environment,
             }
+
+            if "group_states" not in header_data:
+                header_data["group_states"] = None
+            try:
+                group_states = decode_optional_dict_str(header_data["group_states"])
+            except ValueError:
```

Comment:
How come you are catching and ignoring this error?

Reply:
Just being overcautious so that we don't unintentionally break anything. We should log the error here though.

Reply:
Would it make more sense to let the Exception bubble up instead of logging it?

Reply:
I don't want to end up skipping post process if this breaks for some reason. I think that once we're happy that this is working well we can remove the try/except though.

Reply:
I think this try/except is dangerous because if it fails we won't even know about it. If it's supposed to be removed then please add logging to Sentry here and a TODO stating that it is supposed to be temporary for rollout only.

Reply:
Ignore the part about logging, i see you already added that now